### DISCUSSION TITLE
ShowPartiallySelected Functionality disables automatically when I render a counter function in the onChange prop

### DISCUSSION THREAD

**Discussion by Josesan94** [dowjones/react-dropdown-tree-select#557] [Q&A]
Hello there! I've created a function that renders a counter when I select a category. The problem is that when I create the functionality in the onChange prop, the functionality of the automatic selection of parents or the partial selection dissapears too

Here you can see a little video where the functionality shows up and how the automatic selection when click on parents dissapears. Also the partial selection when click on childrens dissapears too. 

I would appreciate the help. Thanks!

**Comment by mrchief** [comment]
Without looking at the code, it's hard to tell. If I have to take a guess, you're using functional components and the re-render is wiping out your state.

**Reply by Josesan94** [reply]
let me know if you have any doubt

**Reply by mrchief** [reply]
I don't see your counter logic in the above link?

**Reply by Josesan94** [reply]
Sorry! My bad.

**Reply by JoseSNubimetrics** [reply]
have you had the chance to take a look? @mrchief

**Reply by mrchief** [reply]
Yeah and the problem is what I suspected. Anytime you use `setXXX` it updates the state and causes a render, so it re-inits the component. You'd need to use the [prevent-re-render-on-parent-propstate-changes](https://github.com/dowjones/react-dropdown-tree-select/blob/develop/docs/HOC.md#prevent-re-render-on-parent-propstate-changes) HOC approach to prevent it. There's also a hooks based approach in [here](https://github.com/dowjones/react-dropdown-tree-select/issues/313).

**Reply by JoseSNubimetrics** [reply]
Thanks, that's very helpful! Resolving that with the HOC approach I will be able to prevent re-render keep the state of the childs when a parent is selected??

**Reply by JoseSNubimetrics** [reply]
I mean, it would be the same if I put an UseEffect hook there, instead of the  componenteWillReceiveProps and ComponentShouldUpdate functions , right?

**Reply by mrchief** [reply]
Have you looked at the hooks based example?

**Reply by JoseSNubimetrics** [reply]
Yes, but I don`t usually work with the older version of react, So I'm not familiarized with react classes, mount and didunmount

**Reply by JoseSNubimetrics** [reply]
I've tried to work with the HOC approach that you make, but Still I can't prevent re-render because childrens don't check when I click on the parent
